generator/restriction_generator.cpp

```diff
@@ -0,0 +1,50 @@
+#include "generator/restriction_generator.hpp"
+#include "generator/restriction_collector.hpp"
+
+#include "coding/file_container.hpp"
+#include "coding/file_writer.hpp"
+
+#include "base/logging.hpp"
+#include "base/stl_helpers.hpp"
+
+#include "defines.hpp"
+
+#include "std/algorithm.hpp"
+
+using namespace feature;
+
+namespace routing
+{
+bool BuildRoadRestrictions(string const & mwmPath, string const & restrictionPath,
+                           string const & featureId2OsmIdsPath)
+{
+  LOG(LINFO,
+      ("BuildRoadRestrictions(", mwmPath, ", ", restrictionPath, ", ", featureId2OsmIdsPath, ");"));
+  RestrictionCollector restrictionCollector(restrictionPath, featureId2OsmIdsPath);
+  if (!restrictionCollector.HasRestrictions() || !restrictionCollector.IsValid())
+  {
+    LOG(LWARNING, ("No valid restrictions for", mwmPath, "It's necessary to check if",
+                   restrictionPath, "and", featureId2OsmIdsPath, "are available."));
+    return false;
+  }
+
+  RestrictionVec const & restrictions = restrictionCollector.GetRestrictions();
+
+  auto const firstOnlyIt =
+      upper_bound(restrictions.cbegin(), restrictions.cend(),
```

Comment:
I bet that you need to ```lower_bound(..., Restriction(Only, {} /* links */))``` instead of what you have written. Otherwise, the name of iterator misleads.